File: Bai_76/main.cpp
#include <fstream>
#include <vector>

std::vector<std::vector<int>> find_subarrays(std::vector<int> arr, int target) {
    std::vector<std::vector<int>> result;
    for (int i = 0; i < arr.size(); i++) {
        int sum = 0;
        for (int j = i; j < arr.size(); j++) {
            sum += arr[j];
            if (sum == target) {
                result.push_back(std::vector<int>(arr.begin() + i, arr.begin() + j + 1));
            } else if (sum > target) {
                break;
            }
        }
    }
    return result;
}

int main() {
    std::ifstream input("DAYCONLT.INP");
    std::ofstream output("DAYCONLT.OUT");
    int n, m;
    input >> n >> m;
    std::vector<int> arr(n);
    for (int i = 0; i < n; i++) {
        input >> arr[i];
    }
    std::vector<std::vector<int>> subarrays = find_subarrays(arr, m);
    if (!subarrays.empty()) {
        for (const auto& subarray : subarrays) {
            for (int num : subarray) {
                output << num << ' ';
            }
            output << '\n';
        }
    } else {
        output << '0';
    }
    return 0;
}

File: Bai_76/main.py
def find_subarrays(arr, target):
    result = []
    for i in range(len(arr)):
        sum = 0
        for j in range(i, len(arr)):
            sum += arr[j]
            if sum == target:
                result.append(arr[i:j+1])
            elif sum > target:
                break
    return result

def solve_problem():
    with open('E:\Project Visual Studio Code\Summary_exercises\Bai_76\DAYCONLT.INP', 'r') as file:
        n, m = map(int, file.readline().split())
        arr = list(map(int, file.readline().split()))
    subarrays = find_subarrays(arr, m)
    with open('E:\Project Visual Studio Code\Summary_exercises\Bai_76\DAYCONLT.OUT', 'w') as file:
        if subarrays:
            for subarray in subarrays:
                file.write(' '.join(map(str, subarray)) + '\n')
        else:
            file.write('0')

solve_problem()